### ISSUE TITLE
Using autocomplete to override methods yields unexpected imports

### ISSUE THREAD

**Issue by tillahoffmann** [microsoft/pylance-release#4575]
# Behaviour

## Expected vs. Actual

Expected: using autocomplete leaves imports unchanged if they are already present.
Actual: using autocomplete modifies imports, often leading to invalid Python code.

## Steps to reproduce:

Create a new python file with the following content:

```python
# playground/auto-import.py
class Foo:
    pass


class Bar:
    def foo() -> Foo:
        return Foo()


class Baz(Bar):
    def fo<hit autocomplete here>
```

I expected no imports to be added, but `from playground.auto-import import Foo` is added to the top of the file. 

```python
# playground/auto-import.py after using autocomplete to override `foo` in `Baz`.
from playground.auto-import import Foo


class Foo:
    pass


class Bar:
    def foo() -> Foo:
        return Foo()


class Baz(Bar):
    def foo() -> Foo:
        pass
```

Using autocomplete multiple times will repeatedly add the same line to the top of the file.

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.0
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Pyenv
-   Value of the `python.languageServer` setting: Default

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

```
2023-06-29 19:14:28.429 [info] LINTING: Skipping linting from Python extension, since Flake8 extension is installed and enabled.
2023-06-29 19:14:28.761 [info] Discover tests for workspace name: [redacted] - uri: /[redacted]
2023-06-29 19:14:28.764 [info] > ~/.pyenv/versions/3.10.0/envs/[redacted]/bin/python ~/.vscode/extensions/ms-python.python-2023.10.1/pythonFiles/testing_tools/run_adapter.py discover pytest -- --rootdir . -s --cache-clear
2023-06-29 19:14:28.764 [info] cwd: .
```

</p>
</details>

<details>

<summary>User Settings</summary>

<p>

```

languageServer: "Pylance"

linting
• flake8Enabled: true

testing
• pytestEnabled: true

```

</p>
</details>

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2023.7.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023721-12-July-2023-PreRelease)

**Comment by tillahoffmann** [comment]
Excellent, thank you!
